src/tm-threads.c

```diff
@@ -427,7 +427,11 @@ static void *TmThreadsSlotVar(void *td)
 
     StatsSetupPrivate(tv);
 
-    TmThreadsSetFlag(tv, THV_INIT_DONE);
+    // Each 'worker' thread uses this func to process/decode the packet read.
+    // Each decode method is different to receive methods in that they do not
+    // enter infinite loops. They use this as the core loop. As a result, at this
+    // point the worker threads can be considered both initialized and running.
+    TmThreadsSetFlag(tv, THV_INIT_DONE | THV_RUNNING);
```

Comment:
In this exact instance you are correct, they mean the same thing for worker threads. The worker threads do not enter an infinite loop, thus ``INIT_DONE=RUNNING`` in this instance. The ``RUNNING`` is designed for the init thread to ensure all threads are fully operational, not only initialised. Mainly to ensure all PPT threads are in a ``RUNNING`` state i.e. just before entering their ``while()``.

Reply:
The steps between `INIT_DONE` and `RUNNING` are not part of the thread initialization proper.

My observation is that since both `RUNNING` and `INIT_DONE` are set at the same time, there's no distinction between the two states. If there were, `RUNNING` would be deferred until the thread is fully operational.

Reply:
Yes for the worker threads that is the case. However if you see any of the packet processing threads/methods they set``RUNNING`` later in the process. The main aim here is to ensure they are fully operational. And to ensure they are in a capturing state before notifying suricata is operational.